codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpBoundProtocolGenerator.kt

```diff
@@ -865,15 +864,15 @@ private class ServerHttpBoundProtocolTraitImplGenerator(
                                 if (memberShape.hasTrait<EnumTrait>()) {
                                     rustTemplate(
                                         """
-                                        let v = <#{memberShape}>::try_from(#{PercentEncoding}::percent_decode_str(&v).decode_utf8()?.as_ref())?;
```

Comment:
Although not mentioned in the issue, this seems like the same class of redundancy?

Reply:
Same as https://github.com/awslabs/smithy-rs/pull/1417/files#diff-da9c47cae3f2f75fc01b3625ee430f56f2f495870a77302c3c148e98f7341bd4L876